analytic_engine/src/payload.rs

```diff
@@ -163,10 +176,21 @@ pub enum ReadPayload {
 }
 
 impl ReadPayload {
-    fn decode_write_from_pb(buf: &[u8]) -> Result<Self> {
+    fn decode_write_from_pb(schema: Schema, buf: &[u8]) -> Result<Self> {
```

Comment:
Seem schema is just used in the `Columnar` path, how about pass the reference and just do the late clone in related path?

Reply:
👍